### PR TITLE
Migrate to GitHub actions

### PR DESCRIPTION
**Pull request by xaviarias**
### What does this PR do?

Uses GitHub actions for CI instead of Travis. 

### Where should the reviewer start?

The gradle.yml workflow file.

### Why is it needed?

Travis has become a payment platform.

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Run integration tests
+        run: ./gradlew integrationTest
```
Comment (xaviarias) [inline .github/workflows/gradle.yml:30]: @AlexandrouR what would be the command for running ITs?

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        run: ./gradlew check publish bintrayUpload -Pversion=${GITHUB_REF#'release/'}
+        if: contains(github.ref, '/release/')
+      - name: Release
+        run: ./gradlew closeAndReleaseRepository
+        if: contains(github.ref, '/release/')
+      - name: GitHub release
+        run: ./scripts/github-release.sh
+        if: contains(github.ref, '/release/')
+      - name: Javadoc release
+        run: ./gradlew -Dorg.ajoberstar.grgit.auth.username=${{ secrets.GITHUB_TOKEN }} core:gitPublishPush
```
Comment (SweeXordious) [inline .github/workflows/gradle.yml:73]: Are we releasing java docs only for `core` ? Why not release the docs for all the library ?

**Review by SweeXordious — APPROVED** [review]
Great work !

File: .github/workflows/gradle.yml (hunk 2 above)
Comment (xaviarias) [reply .github/workflows/gradle.yml:73]: Not sure about that, I think for the moment only core.

File: .github/workflows/gradle.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    env:
+      OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+      OSSRH_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+      GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+      BINTRAY_USER: ${{ secrets.BINTRAY_USER }}
+      BINTRAY_KEY: ${{ secrets.BINTRAY_KEY }}
+      GITHUB_PERSONAL_ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
```
[the remark this answers is not in the document]
Comment (xaviarias) [reply .github/workflows/gradle.yml:42]: That would be awesome, we can share actions and also workflows. There is already a repo for the workflows [here](https://github.com/web3j/.github/tree/main/workflow-templates).

For the secret keys, there we can define them at organisation level and they will be seen from all projects.

For the beginning I wanted to have GH actions working for Web3j and then refine the config.

File: .github/workflows/gradle.yml (hunk 2 above)
Comment (AlexandrouR) [reply .github/workflows/gradle.yml]: yes we are releasing the docs only for core

**Review by AlexandrouR — APPROVED** [review]
LGTM

**Review by kare — COMMENTED** [review]
Hey @SweeXordious @xaviarias,

I'm working on a similar Travis CI to Github Actions conversion. I'm new to Github Actions. I hope there is something useful for you guys:
https://github.com/streamr-dev/engine-and-editor/pull/897/files
https://github.com/streamr-dev/engine-and-editor/pull/899/files

I think (at the moment) that it's a good idea to define exact versions for Github Actions (such as: `actions/setup-java@v1.4.3`) and the use Dependabot to send PR's about updated actions.

**Pull request merged**